reposettings.py

```diff
@@ -169,6 +169,41 @@ def set(repo: Repository.Repository, config):
                 newsettings['dismiss_stale_reviews'] = bool(rules['dismiss-stale-reviews'])
             if 'required-review-count' in rules:
                 newsettings['required_approving_review_count'] = int(rules['required-review-count'])
+            if 'dismiss-pull-request-reviews' in rules:
+                if 'users' in rules['dismiss-pull-request-reviews']:
+                    newsettings['dismissal_users'] = rules['dismiss-pull-request-reviews']['users']
+                if 'teams' in rules['dismiss-pull-request-reviews']:
+                    newsettings['dismissal_teams'] = rules['dismiss-pull-request-reviews']['teams']
+                if 'apps' in rules['dismiss-pull-request-reviews']:
+                    newsettings['dismissal_apps'] = rules['dismiss-pull-request-reviews']['apps']
+            if 'allow-bypass-pull-request-reviews' in rules:
+                if 'users' in rules['allow-bypass-pull-request-reviews']:
+                    newsettings['users_bypass_pull_request_allowances'] = rules['allow-bypass-pull-request-reviews']['users']
```

Comment:
The inconsistency between what we have in our YAML and what we send to GitHub is because the API ask for `users_bypass_pull_request_allowances` while the UI has this:

![GitHub screenshot where you can read: Allow specified actors to bypass required pull requests. Specify people, teams, or apps who are allowed to bypass required pull requests.](https://user-images.githubusercontent.com/53659978/234208232-fe18645e-78c1-4c70-ad77-2d2be6dbc3be.png)

I preferred to abstract the user from the API and be more close to the UI.

Reply:
This is one of those tricky situations where we have to choose between the more obvious and the more consistent approach.

I think more consistent is better here, if the rest of our settings are like github's API names, changing a few of them would likely feel unexpected to users.

Reply:
> The inconsistency between what we have in our YAML and what we send to GitHub

That is what we send to PyGithub, which could even be slightly different from what is finally sent in the corresponding request payload... Therefore I also think that keeping it closer to the UI is a better option here (let's say that it is _less confusing_).

Reply:
From my point of view, we are an abstraction of the API. We might have to document our options for users more than be close to API things.

We can take care of it in the next iteration.

Reply:
Solved in [#0e24cca](https://github.com/txqueuelen/reposettings/pull/11/commits/0e24ccaecb00a4d1f421b4f42ec3d8aa6c27b704)